run-clang-format.py

```diff
@@ -298,10 +309,26 @@ def main():
         return ExitStatus.TROUBLE
 
     retcode = ExitStatus.SUCCESS
+
+    excludes = []
```

Comment:
Sorry for being so painful.
I'm not sure if it's worth it catching the exception here and pretty-print it, because qui a few other places do not.
I catch mostly the expected errors.

Also, I would put the try/except the closest as possible to the code that raise it.

In `get_clang_format_ignore_excludes()`:

```python
def excludes_from_file(ignore_file):
    excludes = []
    try:
        with io.open(ignore_file, 'r', encoding='utf-8') as f:
            for line in f:
                if line.startswith('#'):
                    # ignore comments
                    continue
                pattern = line.rstrip()
                if not pattern:
                    # allow empty lines
                    continue
                excludes.append(pattern)
    except EnvironmentError as e:
        if e.errno != errno.ENOENT:
            raise
    return excludes;

```

Reply:
Np, sorry for the misunderstanding. I did implement it like that at first but when I saw your error reporting in the main function I thought I need to refactor it like that.
Thank you.